sql-to-dbsp-compiler/lib/sqllib/src/binary.rs

```diff
@@ -52,4 +54,41 @@ impl ByteArray {
     pub fn new(d: &[u8]) -> Self {
         Self { data: d.to_vec() }
     }
+
+    pub fn length(self: &Self) -> usize {
+        self.data.len()
+    }
+
+    pub fn zip<F>(self: &Self, other: &Self, op: F) -> ByteArray
+        where F: Fn(&u8, &u8) -> u8
+    {
+        let self_len = self.data.len();
+        let other_len = other.data.len();
+        if self_len != other_len {
+            panic!("Cannot operate on BINARY objects of different sizes {} and {}",
```

Comment:
Wouldn't this kill the entire pipeline?

Reply:
Right, that's https://github.com/feldera/feldera/issues/297
We don't have any solution right now.
This is the semantics of SQL.